sdk/resource/resource.go

```diff
@@ -40,7 +40,7 @@ var (
 			otel.Handle(err)
 		}
 		return r
-	}(Detect(context.Background(), defaultServiceNameDetector{}, TelemetrySDK{}))
+	}(Detect(context.Background(), FromEnv{}, defaultServiceNameDetector{}, TelemetrySDK{}))
```

Comment:
```suggestion
	}(Detect(context.Background(), defaultServiceNameDetector{}, FromEnv{}, TelemetrySDK{}))
```

The `defaultServiceNameDetector` shouldn't override a service name specified in the environment.

Reply:
I will make this change